### ISSUE TITLE
 Test Execution Issues

### ISSUE THREAD

**Issue by mikaayil** [narendraj9/hledger-mode#56]
Hello,

I initiated Emacs in a clean state using `emacs -Q`. While executing the test cases, I observed that not all of them passed as expected.  

@shivjm: Can you provide some insights regarding these errors? (#53)

```
Selector: t
Passed:  9
Failed:  5 (5 unexpected)
Skipped: 0
Total:   14/14

Started at:   2023-09-10 17:21:16+0200
Finished.
Finished at:  2023-09-10 17:21:16+0200

F.........FFFF

F ert-hledger-account-bounds-is-correct
    (void-function hledger-bounds-of-account-at-point)

F hledger-accounts-should-not-update
    With ‘hledger-invalidate-completions’ set to ‘on-idle’, saving or
    (void-function hledger-mode)

F hledger-accounts-should-update-on-edit
    With ‘hledger-invalidate-completions’ set to ‘(on-idle
    (void-function hledger-mode)

F hledger-accounts-should-update-on-save
    With ‘hledger-invalidate-completions’ set to ‘(on-idle
    (void-function hledger-mode)

F hledger-date-manipulation-works
    (void-function hledger-add-days-to-entry-date)
```

**Comment by shivjm** [comment]
Thanks for bringing this to my attention. May I ask which version of Emacs and which revision of hledger-mode you’re using? With Emacs 29 and hledger-mode from `master`, I do currently see an error saying `hledger-bounds-of-account-at-point` doesn’t exist in `ert-hledger-account-bounds-is-correct`, which is confusing to me since the function exists and the tests passed earlier, and another in `hledger-date-manipulation-works` because—as I realized only recently—`iso8601-parse` doesn’t exist in Emacs 29, or at least not natively.

I’ll investigate the two errors I encountered, at least. I apologize for the confusion!

**Comment by mikaayil** [comment]
Of course. I'm using Emacs 29.1, and I also ran the tests from the master branch with the latest commit (ae551c0601dc39e0f751fe52a97fab01f4844e0a)

**Comment by shivjm** [comment]
I’m sorry for taking so long to respond. I tried running from `master` and, apart from the issue with `iso8601-parse` not being available (addressed by #61), it seems like the others are related to load order in some way that I’m unable to identify. They only occur sporadically for me in an <kbd>emacs -Q</kbd>. Would it be possible for you to test whether they still occur with the current `master`?
